application/models/TokenDynamic.php

```diff
@@ -1079,4 +1063,30 @@ public function getSurveyId()
     {
         return self::$sid;
     }
+
+    /**
+     * Returns true if the token can be used
+     * @return bool
+     */
+    public function canBeUsed()
+    {
+        return !empty($this->token)
+            && (
+                $this->completed == "N"
+                || empty($this->completed)
+                || $this->survey->isAllowEditAfterCompletion
+            );
+    }
+
+    public function canBeEmailed()
+    {
+        return !empty($this->token)
+            && $this->emailstatus == "OK"
+            && $this->email
+            && $this->completed == "N"
+            && (
```

Comment:
replace by && $this->canBeused()

Reply:
As to avoid complexity, and release the fix, i will do that on another ticket.
Also, please bear in mind the usesLeft on mailed but not on canBeused

Reply:
Created this: https://bugs.limesurvey.org/view.php?id=18993

Reply:
> Also, please bear in mind the usesLeft on mailed but not on canBeused

Right, but to be sended : it seems clar it must be used ;) : and if useLeft > 0 : can be used , no ?